pkg/controller/registry/resolver/resolver.go

```diff
@@ -568,6 +595,16 @@ func (r *SatResolver) addInvariants(namespacedCache cache.MultiCatalogOperatorFi
 			}
 			packageConflictToInstallable[prop.PackageName] = append(packageConflictToInstallable[prop.PackageName], installable.Identifier())
 		}
+
+		// apply runtime constraints to packages that aren't already installed
+		if !catalog.Virtual() && r.runtimeConstraintsProvider != nil {
```

Comment:
I'm actually fine with this AddInvariants solution. I was expecting to be messier but it looks reasonable to me.